app/scripts/controllers/network/createInfuraClient.test.js

```diff
@@ -0,0 +1,232 @@
+      'eth_getTransactionByBlockHashAndIndex',
+    );
+  });
+
+  describe('eth_getTransactionByBlockNumberAndIndex', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getTransactionByBlockNumberAndIndex',
+    );
+  });
+
+  describe('eth_getTransactionByHash', () => {
+    testsForRpcMethodsThatCheckForBlockHashInResponse(
+      'eth_getTransactionByHash',
+    );
+  });
+
+  describe('eth_getTransactionReceipt', () => {
+    testsForRpcMethodsThatCheckForBlockHashInResponse(
+      'eth_getTransactionReceipt',
+    );
+  });
+
+  describe('eth_getUncleByBlockHashAndIndex', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getUncleByBlockHashAndIndex',
+    );
+  });
+
+  describe('eth_getUncleByBlockNumberAndIndex', () => {
+    testsForRpcMethodThatDoesNotSupportParams(
+      'eth_getUncleByBlockNumberAndIndex',
+    );
+  });
+
+  describe('eth_getUncleCountByBlockHash', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_getUncleCountByBlockHash');
+  });
+
+  describe('eth_getUncleCountByBlockNumber', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_getUncleCountByBlockNumber');
+  });
+
+  describe('eth_protocolVersion', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_protocolVersion');
+  });
+
+  describe('shh_version', () => {
+    testsForRpcMethodThatDoesNotSupportParams('shh_version');
+  });
+
+  describe('test_blockCache', () => {
+    testsForRpcMethodThatDoesNotSupportParams('test_blockCache');
+  });
+
+  describe('test_forkCache', () => {
+    testsForRpcMethodThatDoesNotSupportParams('test_forkCache');
+  });
+
+  describe('test_permaCache', () => {
+    testsForRpcMethodThatDoesNotSupportParams('test_permaCache');
+  });
+
+  describe('web3_clientVersion', () => {
+    testsForRpcMethodThatDoesNotSupportParams('web3_clientVersion');
+  });
+
+  describe('web3_sha3', () => {
+    testsForRpcMethodThatDoesNotSupportParams('web3_sha3');
+  });
+
+  // == RPC methods that support multiple params (including a "block" param)
+  //
+  // RPC methods in this category take a non-empty `params` array, and more
+  // importantly, one of these items can specify which block the method applies
+  // to. This block param may either be a tag ("earliest", "latest", or
+  // "pending"), or a specific block number; or this param may not be
+  // provided altogether, in which case it defaults to "latest". Also,
+  // "earliest" is just a synonym for "0x00".
+  //
+  // The fact that these methods support arguments affects the caching strategy,
+  // because if two requests are made with the same method but with different
+  // arguments, then they will be cached separately. Also, the block param
+  // changes the caching strategy slightly: if "pending" is specified, then the
+  // request is never cached.
+
+  describe('eth_getBlockByNumber', () => {
+    testsForRpcMethodThatSupportsMultipleParams('eth_getBlockByNumber', {
+      numberOfParams: 0,
```

Comment:
Curiously, this method has two required parameters, but the first one is the block parameter.

Saying that the "number of parameters" is zero here is a bit confusing. I'm guessing this was set to zero because this method breaks with the convention of the block parameter being last.

Reply:
This is a bit confusing for the methods that support a blockRef, even in cases where it is last, because the true "number of parameters" is still wrong. e.g. `eth_call` is set as supporting one parameter, but it really has two including the block parameter.

Reply:
Similarly, methods that support a block parameter maybe shouldn't be described as having no parameters (e.g. `eth_getBlockTransactionCountByHash`)

Reply:
Ah, yes, this should probably be more like `blockRefParamIndex`. I can fix this.

Reply:
Fixed in https://github.com/MetaMask/metamask-extension/pull/15556/commits/f815a103f8bf7077869f8119365e2d14fc50d4ed.